File: nexus-bank-ui/src/services/transactionService.ts
import api from './api';
import type { Transaction, BeneficiaryInfo } from '../types';

export const transactionService = {
  async getHistory(): Promise<Transaction[]> {
    const response = await api.get<Transaction[]>('/transactions/history');
    return response.data;
  },

  async resolveBeneficiary(bankCode: String, accountNumber: String): Promise<BeneficiaryInfo> {
    const response = await api.get<BeneficiaryInfo>('/external/nibss/resolve', {
      params: { bankCode, accountNumber }
    });
    return response.data;
  },

  async transfer(request: any): Promise<Transaction> {
    const response = await api.post<Transaction>('/transactions/transfer', request);
    return response.data;
  },

  async deposit(request: any): Promise<Transaction> {
    const response = await api.post<Transaction>('/transactions/deposit', request);
    return response.data;
  },

  async withdraw(request: any): Promise<Transaction> {
    const response = await api.post<Transaction>('/transactions/withdraw', request);
    return response.data;
  }
};

File: src/main/java/com/nexus/core/transaction/TransactionService.java
package com.nexus.core.transaction;

import com.nexus.core.common.TransactionRequest;
import com.nexus.core.common.TransactionResponse;
import java.util.List;
import java.util.UUID;

public interface TransactionService {
    TransactionResponse transfer(TransactionRequest request);
    TransactionResponse deposit(TransactionRequest request);
    TransactionResponse withdraw(TransactionRequest request);
    List<TransactionResponse> getAccountTransactions(String accountNumber);
    TransactionResponse getTransactionById(UUID id);
}
